module/zfs/zio_inject.c

```diff
@@ -148,6 +156,7 @@ zio_match_handler(const zbookmark_phys_t *zb, uint64_t type,
 	    zb->zb_level == record->zi_level &&
 	    zb->zb_blkid >= record->zi_start &&
 	    zb->zb_blkid <= record->zi_end &&
+	    (record->zi_dvas == 0 || (record->zi_dvas & (1 << dva))) &&
```

Comment:
It'd be nice if the passed `dva` was still a bitmask so it could be directly compared without the addition conditionals. 

```
        dvas & record->zi_dvas &&
```

Reply:
I know its a bit uglier, but I like it more this way. To me once is a list of possibilities (the bitmask) and the other is one of the possibilities. Eve though its a bit longer I think it makes more sense like this. If you feel strongly about it though I can change it.